ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -612,6 +617,48 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (*api.IPFSRepoStat, error)
 	return &stats, nil
 }
 
+// RepoGC performs a garbage collection sweep on the repo.
+func (ipfs *Connector) RepoGC(ctx context.Context) (*api.RepoGC, error) {
+	ctx, span := trace.StartSpan(ctx, "ipfsconn/ipfshttp/RepoGC")
+	defer span.End()
+
+	id, err := ipfs.ID(ctx)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+	repoGC := api.RepoGC{
+		Peer: id.ID,
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
+	defer cancel()
+
+	res, err := ipfs.doPostCtx(ctx, ipfs.client, ipfs.apiURL(), "repo/gc", "", nil)
+	if err != nil {
+		logger.Error(err)
+		return &repoGC, err
+	}
+	defer res.Body.Close()
+
+	dec := json.NewDecoder(res.Body)
+	for {
+		resp := ipfsRepoGCResp{}
+		if err := dec.Decode(&resp); err == io.EOF {
+			break
+		} else if err != nil {
+			return &repoGC, err
+		}
+		if resp.Error != "" {
+			logger.Error("Error while repo gc: ", resp.Error)
+			repoGC.Error += ", " + resp.Error
```

Comment:
put errors in a slice and join them later (by tagging them with the cid they belong to if present `<cid>: <error>, <cid2>: <error2>`? This will result in `, error 1`.

Reply:
Have changed the struct for repo gc. With this we would be able to tag cids with errors.
```
// IPFSRepoGC represents the streaming response sent from ipfs' repo gc API.
type IPFSRepoGC struct {
	Key   cid.Cid `json:"key" codec:"k"`
	Error string  `json:"error,omitempty" codec:"e,omitempty"`
}

// RepoGC contains gc'ed CIDs from a cluster peer's IPFS daemon.
type RepoGC struct {
	Peer  peer.ID      `json:"peer" codec:"p,omitempty"` // the Cluster peer ID
	Keys  []IPFSRepoGC `json:"keys" codec:"k"`
	Error string       `json:"error,omitempty" codec:"e,omitempty"`
}
```